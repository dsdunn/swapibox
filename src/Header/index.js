import React from 'react';
import PropTypes from 'prop-types';

const Header = (props) => { 
  return (
    <div className="header">
      <h1 className="title">SwapiBox</h1>
      <button onClick={props.getPeople} className="people">People</button>
      <button onClick={props.getVehicles} className="vehicles">Vehicles</button>
      <button onClick={props.getPlanets} className="planets">Planets</button>
      <button onClick={props.showFavorites} className="favorites">favorites</button>
    </div>
    )
}

Header.propTypes = {
  getPeople: PropTypes.func,
  getVehicles: PropTypes.func,
  getPlanets: PropTypes.func,
  showFavorites: PropTypes.func
}

export default Header;