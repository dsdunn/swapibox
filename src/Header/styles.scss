.header {
  background-color: #1c1c12;;
  margin-left:250px;
  text-align: center;
}

.controls {
  display: flex;
  justify-content: space-around;
}

h1 {
  color: yellow;
}

.controls button {
  height: 40px;
  color: yellow;
  background-color: darkgray;
  width: auto;
  font-size: 1.7em;
}