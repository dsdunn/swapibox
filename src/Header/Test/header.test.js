import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import Header from '../';

describe('Header', () => {
  let wrapper;
  const mockProps = {
    getPeople: jest.fn(),
    getVehicles: jest.fn(),
    getPlanets: jest.fn(),
    showFavorites: jest.fn()
  }

  it('should match the snapshot', () => {
    wrapper = shallow(<Header props={mockProps} />)
    expect(wrapper).toMatchSnapshot();
  })
})