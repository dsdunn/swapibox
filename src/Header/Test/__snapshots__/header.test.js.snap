// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    props={
        Object {
            "getPeople": [Function],
            "getPlanets": [Function],
            "getVehicles": [Function],
            "showFavorites": [Function],
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="title"
>
          SwapiBox
</h1>,
        <button
          className="people"
          onClick={undefined}
>
          People
</button>,
        <button
          className="vehicles"
          onClick={undefined}
>
          Vehicles
</button>,
        <button
          className="planets"
          onClick={undefined}
>
          Planets
</button>,
        <button
          className="favorites"
          onClick={undefined}
>
          favorites
</button>,
      ],
      "className": "header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "SwapiBox",
          "className": "title",
        },
        "ref": null,
        "rendered": "SwapiBox",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "People",
          "className": "people",
          "onClick": undefined,
        },
        "ref": null,
        "rendered": "People",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Vehicles",
          "className": "vehicles",
          "onClick": undefined,
        },
        "ref": null,
        "rendered": "Vehicles",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Planets",
          "className": "planets",
          "onClick": undefined,
        },
        "ref": null,
        "rendered": "Planets",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "favorites",
          "className": "favorites",
          "onClick": undefined,
        },
        "ref": null,
        "rendered": "favorites",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="title"
>
            SwapiBox
</h1>,
          <button
            className="people"
            onClick={undefined}
>
            People
</button>,
          <button
            className="vehicles"
            onClick={undefined}
>
            Vehicles
</button>,
          <button
            className="planets"
            onClick={undefined}
>
            Planets
</button>,
          <button
            className="favorites"
            onClick={undefined}
>
            favorites
</button>,
        ],
        "className": "header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "SwapiBox",
            "className": "title",
          },
          "ref": null,
          "rendered": "SwapiBox",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "People",
            "className": "people",
            "onClick": undefined,
          },
          "ref": null,
          "rendered": "People",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Vehicles",
            "className": "vehicles",
            "onClick": undefined,
          },
          "ref": null,
          "rendered": "Vehicles",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Planets",
            "className": "planets",
            "onClick": undefined,
          },
          "ref": null,
          "rendered": "Planets",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "favorites",
            "className": "favorites",
            "onClick": undefined,
          },
          "ref": null,
          "rendered": "favorites",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
